### PR TITLE
Query the latest 80 live submissions the first time get into the live submissions pages

### PR DESCRIPTION
**Pull request by Wally5077**
resolve: #153 

**Before live submissions pushed**:
image

**After live submissions pushed**:
image

### CODE REVIEW

File: src/services/LiveSubmissionsService.js [hunk 1]
```diff
@@ -4,17 +4,24 @@ import VerdictIssuedEvent from "../models/VerdictIssuedEvent";
 
 export class LiveSubmissionsService extends AbstractService {
 
-    constructor(studentService, problemService, stompClient) {
+    constructor(studentService, problemService, submissionService, stompClient) {
         super({
             baseURL: process.env.REACT_APP_PROBLEM_SVC_BASE_URL,
```
Comment (nyngwang) [inline src/services/LiveSubmissionsService.js:9]: Should it be `REACT_APP_SUBMISSION_SVC_BASE_URL` like you did in the `src/services/SubmissionService.js`?

File: src/services/LiveSubmissionsService.js [hunk 2]
```diff
@@ -4,17 +4,24 @@ import VerdictIssuedEvent from "../models/VerdictIssuedEvent";
...
         this.problemService = problemService;
+        this.submissionService = submissionService;
         this.stompClient = stompClient;
     }
 
+    async queryExamLatestSubmissions(examId) {
+        let queryParameters = {examId, sortBy: 'submissionTime', ascending: false};
+        return await this.submissionService.getSubmissions(queryParameters)
+            .then(submissions => Promise.all(submissions.map(submission => this.completeLiveSubmissionFields(submission))));
+    }
```
Comment (nyngwang) [inline src/services/LiveSubmissionsService.js:23]: The nested `Promise.all` seems strange, would it possible to re-write it as `Promise.all(arr)` where you put all you need to wait into a single `arr`?

**Review by nyngwang — CHANGES_REQUESTED** [review]
Review Done.

File: src/app/exam/submissions/ExamLiveSubmissions.js [hunk 3]
```diff
@@ -32,6 +33,14 @@ const ExamLiveSubmissions = function () {
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
+    function fetchLiveSubmissions(examId) {
```
Comment (Johnny850807) [inline src/app/exam/submissions/ExamLiveSubmissions.js:36]: Ubiquitous language: rename to `queryExamLatestSubmissions`

File: src/services/LiveSubmissionsService.js (hunk 1 above)
Comment (Johnny850807) [reply src/services/LiveSubmissionsService.js:9]: Great catch

File: src/services/LiveSubmissionsService.js [hunk 4]
```diff
@@ -4,17 +4,24 @@ import VerdictIssuedEvent from "../models/VerdictIssuedEvent";
...
             timeout: 10000,
             tokenSupplier: studentService.currentToken
         });
         this.studentService = studentService;
         this.problemService = problemService;
+        this.submissionService = submissionService;
         this.stompClient = stompClient;
     }
 
+    async queryExamLatestSubmissions(examId) {
```
Comment (Johnny850807) [inline src/services/LiveSubmissionsService.js:19]: rename to `queryLatestExamSubmissions`

File: src/services/LiveSubmissionsService.js [hunk 5]
```diff
@@ -4,17 +4,24 @@ import VerdictIssuedEvent from "../models/VerdictIssuedEvent";
...
+        let queryParameters = {examId, sortBy: 'submissionTime', ascending: false};
+        return await this.submissionService.getSubmissions(queryParameters)
+            .then(submissions => Promise.all(submissions.map(submission => this.completeLiveSubmissionFields(submission))));
```
Comment (Johnny850807) [inline src/services/LiveSubmissionsService.js:22]: I think the correct way is as below
```js
const submissions = await this.submissionService.getSubmissions(queryParameters);
return Promise.all(submissions.map(submission => this.completeLiveSubmissionFields(submission)));
```

Comment (Johnny850807) [reply src/services/LiveSubmissionsService.js:22]: BTW, this is elegant, cool.

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done!

**Review by Johnny850807 — APPROVED** [review]
LGTM

**Review by nyngwang — APPROVED** [review]
LGTM.

**Review by Chaoyuuu — APPROVED** [review]
LGTM

**Review by timt1028 — APPROVED** [review]
LGTM

**Pull request merged**